RetailCoder.VBE/UI/Refactorings/AssignedByValParameterQuickFixDialog.cs

```diff
@@ -1,22 +1,22 @@
 ﻿using System;
-using System.Linq;
 using System.Windows.Forms;
 using Rubberduck.Parsing.Symbols;
 using Rubberduck.Inspections;
+using Rubberduck.Inspections.QuickFixes;
+using Rubberduck.VBEditor;
 
 namespace Rubberduck.UI.Refactorings
 {
     public partial class AssignedByValParameterQuickFixDialog : Form, IDialogView
     {
-        private readonly string[] _procedureLines;
-        private bool _userInputIsValid;
+        private AssignedByValParameterQuickFixHelper _quickFixHlpr;
 
-        public AssignedByValParameterQuickFixDialog(string[] procedureLines)
+        public AssignedByValParameterQuickFixDialog(Declaration target, QualifiedSelection selection)
         {
-            _procedureLines = procedureLines;
-            _userInputIsValid = false;
             InitializeComponent();
             InitializeCaptions();
+            Target = target;
+            _quickFixHlpr = new AssignedByValParameterQuickFixHelper(target, selection);
```

Comment:
The helper is used in the dialog for the sole purpose of supporting dynamic evaluation of the user-entered variable name.  So, rather than have the dialog call back to the QuickFix class directly for this validation support, I chose to have it use the helper class.  Another option is to pass the dialog the list of 'forbidden' names (in the ctor or as a property) and remove the helper altogether (I'm thinking I like the last option best).